addons/website/models/website.py

```diff
@@ -1312,12 +1312,13 @@ def _get_canonical_url_localized(self, lang, canonical_params):
 
             router = http.root.get_db_router(request.db).bind('')
             path = router.build(rule.endpoint, args)
+            if lang != self.default_lang_id:
+                path = f'/{lang.url_code}{path}'
         except (NotFound, AccessError, MissingError):
             # The build method returns a quoted URL so convert in this case for consistency.
-            path = urls.url_quote_plus(request.httprequest.path, safe='/')
-        lang_path = f'/{lang.url_code}' if lang != self.default_lang_id else ''
+            path = urls.url_quote_plus(request.httprequest.environ['REQUEST_URI'], safe='/')
```

Comment:
This require you to upgrade Werkzeug from 0.14.1 to 0.16.1.
This is fine, as this is what is required in the requirements.txt, but just to let people know in case someone ends up here when getting the error.